packages/overlay/src/integrations/sentry/components/traces/spans/SpanItem.tsx

```diff
@@ -51,10 +51,8 @@ const SpanItem = ({
       setSpanNodeWidth(newLeftWidth);
     }
   };
-  const isQueried = useMemo(() => {
-    if (!query) return false;
-    return span.span_id.includes(query) || span.op?.includes(query) || span.description?.includes(query);
-  }, [query, span.span_id, span.op, span.description]);
+
+  const isQueried = !showOnlyMatched && query && matchesQuery(span);
```

Comment:
I think renaming this to `shouldHighlight` makes more sense with the new approach.
```suggestion
  const shouldHighlight = !showOnlyMatched && query && matchesQuery(span);
```